pyspectral/rayleigh.py

```diff
@@ -260,15 +274,15 @@ def get_reflectance(self, sun_zenith, sat_zenith, azidiff, bandname, redband=Non
             if redband is not None:
                 redband = from_array(redband, chunks=redband.shape)
 
-        clip_angle = rad2deg(arccos(1. / sunz_sec_coord.max()))
-        sun_zenith = clip(sun_zenith, 0, clip_angle)
+        sun_zenith = self._clip_angles_inside_coordinate_range(sun_zenith, sunz_sec_coord.max())
         sunzsec = 1. / cos(deg2rad(sun_zenith))
-        clip_angle = rad2deg(arccos(1. / satz_sec_coord.max()))
-        sat_zenith = clip(sat_zenith, 0, clip_angle)
+
+        sat_zenith = self._clip_angles_inside_coordinate_range(sat_zenith, satz_sec_coord.max())
         satzsec = 1. / cos(deg2rad(sat_zenith))
+
         shape = sun_zenith.shape
 
-        if not(wvl_coord.min() < wvl < wvl_coord.max()):
+        if not wvl_coord.min() < wvl < wvl_coord.max():
```

Comment:
Should this be done first in the function, so that the lut doesn't need to be read at all if the wavelength is outside the visible range?

Reply:
Well, that would be nice, but things are quite entangled at the moment with the Dask / non-Dask code. Before returning the zero-array a chunk size is being calculated from the sun-zenith and redband arrays, so they are needed here atm! :-(

There is a need for refactoring indeed. But rather do that in another PR